src/components/grid.js

```diff
@@ -1,58 +1,114 @@
-import React from 'react';
+import React, { Component } from 'react';
 import HotTable from 'react-handsontable';
 
 
-//var hot=null;
-
-class GridComponent extends React.Component {
+class Grid extends Component {
 
   constructor(props) {
     super(props);
-    this.state = { testCount: 0 };
-    this.myRef;
   }
 
 
+  componentWillMount() {
+    console.log('did mount');
+  }
 
-  handleChange(event) {
+  componentDidMount() {
+
+    let Hot = this.tableRef.hotInstance;
+    Hot.updateSettings({
+      afterChange: () => {
+        this.updateDataDump()
+      },
+      afterColumnMove: () => {
+        this.updateDataDump()
+      },
+      afterRemoveRow: () => {
+        this.updateDataDump()
+      },
+      afterRemoveCol: () => {
+        this.updateDataDump()
+      },
+      modifyColWidth: () => {
+        this.updateDataDump()
+      },
+      afterOnCellMouseOver: (event, coords, tableData) => {
+        console.log(coords);
+        this.updateCellMouseOver(coords);
+      },
+
+
+    });
 
-    var Hot = this.tableRef.hotInstance;
-    console.log(Hot);
-    //  var table = Hot.getElementsByTagName('table')[0];
-    var table = Hot.table;
-    console.log(table);
+
+  }
+
+
+
+
+
+
+  shouldComponentUpdate(nextProps, nextState) {
+    // console.log(' shouldComponentUpdate in GRID nextProps...');
+    // console.log(nextProps);
+    return false;
+  }
 
 
+  componentWillUpdate() {
+    //return false;
+  }
 
-    //init
-    var tableJsonOutput = { 'filename': 'abc1234' };
 
-    //tableJsonOutput['data']= this.refs.hot.hotInstance.getData();
+  updateDataDump() {
+    //console.log('update event is fired');
+    //console.log(this);
+    var Hot = this.tableRef.hotInstance;
```

Comment:
Might be worth using the ES6 `const` throughout if we can, or `let` if the variable needs to be mutated.